pkg/payload/payload_test.go

```diff
@@ -123,3 +123,73 @@ func mustRead(path string) []byte {
 	}
 	return data
 }
+
+func TestExclude(t *testing.T) {
+	tests := []struct {
+		exclude     string
+		profile     string
+		annotations map[string]interface{}
+
+		isExcluded bool
+	}{
+		{
+			annotations: map[string]interface{}{},
+		},
+		{
+			exclude:     "",
+			annotations: map[string]interface{}{"exclude.release.openshift.io/identifier": "true"},
+		},
+		{
+			exclude:     "identifier",
+			annotations: map[string]interface{}{"exclude.release.openshift.io/identifier": "true"},
+			isExcluded:  true,
+		},
+		{
+			profile:     "",
+			annotations: map[string]interface{}{"include.release.openshift.io/edge": "true"},
```

Comment:
this is a weird test-case.  I'd expect an empty-string profile to cause `shouldExclude` to return an error (which would require us to update its call signature) or panic or some-such.  If we happen to get into that situation, loading a bunch of competing manifests from different profiles does not seem like something that would work well.

Reply:
Yes it's a problem.
At the initialization of the CVO, it loads the payload to get the verifier ConfigMap (https://github.com/openshift/cluster-version-operator/pull/404/files#diff-85c31beb4341b4c52f892c0581bbb5d6R226) but don't have access to the profile. At this point of the install, the CVO container is mostly alone and no resources are present. Either we wait for the ConfigMap (but it's optional, so how can we do that ?) or we load all or default manifests and take the first verifier.
In the sync loop, it works correctly because the ClusterVersion object acts as a lock and the ConfigMap cluster-profile is created before.

Reply:
>  Either we wait for the ConfigMap (but it's optional, so how can we do that ?) or we load all or default manifests and take the first verifier.

"verifier"?  The enhancement line I comment on [here][1] suggests the no-configured-profile behavior should be "exclude manifests with any profile annotations".  That also mitigates the config-loading race, by pushing no profile-specific manifests until a profile config has been found.

[1]: https://github.com/openshift/enhancements/pull/200#discussion_r457408715

Reply:
The verifier is created at this point using an update: https://github.com/openshift/cluster-version-operator/blob/master/pkg/cvo/cvo.go#L245.
This is done once at the initialization of the CVO, so we can't wait for the profile to appear I suppose.

Reply:
Why would you need profile-specific verification?

Reply:
> This is done once at the initialization of the CVO...

We probably need to cache the profile and reload from disk if it's changed since the last sync cycle.

Reply:
> Why would you need profile-specific verification? 

You're right. I don't want it. 
At initialization, we can load default profile manifests and take the verifier. It removes this empty string profile and simplify `shouldExclude`. I will give it a try a do that.

> We probably need to cache the profile and reload from disk if it's changed since the last sync cycle.

Cache the profile is I think a good thing. Can the informer notify when a change appear?

Reply:
> Can the informer notify when a change appear?

The ConfigMap informer could, but it would make a generic interface/getter more awkward to expose that.  I'd rather just poll once at the start of each sync cycle.